app/controllers/index.rb

```diff
@@ -0,0 +1,14 @@
+enable :sessions
+
+get '/' do
+  @user_id = session[:current_user_id]
+  @username = User.find(@user_id).name if @user_id != nil
+  @logged_in = (@user_id != nil)
+  @error = session[:error]
+  erb :index
```

Comment:
you're checking `@user_id != nil` twice which is a code smell. 

looking up the current user based on the ID in the session should be in a helper method called `current_user`